gtsam/discrete/DiscreteFactorGraph.cpp

```diff
@@ -64,10 +65,17 @@ namespace gtsam {
   }
 
   /* ************************************************************************ */
-  DecisionTreeFactor DiscreteFactorGraph::product() const {
-    DecisionTreeFactor result;
+  TableFactor DiscreteFactorGraph::product() const {
```

Comment:
why is this a TableFactor now? I don't think we want to switch *all* discrete inference to TableFactors